publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationHandler.java

```diff
@@ -238,14 +239,27 @@ private Publication unpublishPublication(UnpublishPublicationRequest unpublishPu
                                                             existingPublication,
                                                             userInstance);
         resourceService.unpublishPublication(updatedPublication);
-
+        setPendingTicketsToNotRelevant(updatedPublication);
         updatedPublication = resourceService.getPublication(updatedPublication);
         persistNotification(updatedPublication);
         updateNvaDoi(updatedPublication);
-
         return updatedPublication;
     }
 
+    private void setPendingTicketsToNotRelevant(Publication updatedPublication) {
+        var tickets = resourceService.fetchAllTicketsForResource(Resource.fromPublication(updatedPublication));
+        tickets.filter(this::ticketIsNoLongerRelevant).forEach(this::setTicketToNoLongerRelevant);
+    }
```

Comment:
Is this the only way to update items?
Fetch -> modify -> persist, over a network?

Reply:
In a NoSQL database, I think so yes.

Reply:
I saw Stig's question as: "is it possible to do a "update ... where ..." which relational databses support?". 
Which gives you an atomic operation with the drawback: you dont know what you just updated unless you do some begin transaction update, read rollback or commit anyways. Some NoSQL such as DynamoDb support transactions. So in the event we see the non-atomic updates become a problem, we could refactor in resourceService to do transactions as Kir suggest further down: make resource service update the tickets when updating the publication.

Reply:
This is fair.

Cluster the updates with `TransactWriteItems` in the service.